echopype/process/process_base.py

```diff
@@ -317,12 +329,12 @@ def _cal_narrowband(self, ed, env_params, cal_params, cal_type,
                                    / (32 * np.pi**2)))
 
             # Calibration and echo integration
-            Sv = ed.raw.backscatter_r + spreading_loss + absorption_loss - CSv - 2 * cal_params['sa_correction']
+            Sv = ed.raw['backscatter_r'] + spreading_loss + absorption_loss - CSv - 2 * cal_params['sa_correction']
             Sv.name = 'Sv'
             Sv = Sv.to_dataset()
 
-            # Attach calculated range into data set
-            Sv['range'] = (('frequency', 'ping_time', 'range_bin'), self._restructure_range(ed))
+            # Attach calculated range (with units meter) into data set
+            Sv = Sv.merge(ed.range)
 
             # Save Sv into the calling instance and
```

Comment:
The following error occurs during `io.save_file` if the `ping_time `changes.
`ValueError: Zarr requires uniform chunk sizes except for final chunk. Variable named 'Sv' has incompatible dask chunks ...  Consider rechunking using chunk().`